<launch>
    <arg name="teleop" default="True" />
    <arg name="Gazebo" default="True" />
    <arg name="slam_toolbox" default="True" />

    <include file="$(find turtlebot3_bringup)/launch/turtlebot3_remote.launch" />

    <!-- turtlebot3_teleop -->
    <group if="$(arg teleop)">
        <node pkg="turtlebot3_teleop" type="turtlebot3_teleop_key" name="turtlebot3_teleop_keyboard" output="screen"></node>
    </group>

    <!-- slam_toolbox -->
    <group if="$(arg slam_toolbox)">
		<node pkg="slam_toolbox" type="sync_slam_toolbox_node" name="slam_toolbox" output="screen">
			<rosparam command="load" file="$(find slam_toolbox)/config/mapper_params_online_sync.yaml" />
		</node>
	</group>

    <!-- Gazebo -->
    <group if="$(arg Gazebo)">
        <include file="$(find turtlebot3_gazebo)/launch/turtlebot3_house.launch" />
    </group>

    <!-- turtlebot3_navigation -->
    <include file="$(find turtlebot3_navigation)/launch/move_base.launch" />
    <node pkg="rviz" type="rviz" name="rviz" required="true" args="-d $(find turtlebot3_navigation)/rviz/turtlebot3_navigation.rviz"/>
    

</launch>
